ext/symengine/symengine_utils.c

```diff
@@ -59,10 +61,16 @@ void sympify(VALUE operand2, basic_struct *cbasic_operand2) {
             break;
 
         case T_DATA:
+            c = rb_obj_classname(operand2);
+            if(strcmp(c, "BigDecimal") == 0){
+                c = RSTRING_PTR( rb_funcall(operand2, rb_intern("to_s"), 1, rb_str_new2("F")) );
+                real_mpfr_set_str(cbasic_operand2, c, 200);
```

Comment:
I tried having `#ifdef HAVE_SYMENGINE_MPFR` , but it does not reach the code even though the symengine was built with MPFR and MPC. How can I find whether symengine has been build with MPFR and/or MPC.

Also, how can I translate that into ruby, because I would need to have a way to check that in the spec as well.

Can we have a function in the cwrapper, which returns true or false, depending on the flag? Then have a Ruby Wrapper for that, to access it from Ruby?

This is the solution I am currently thinking of. What do you think?

@isuruf @abinashmeher999